Added Custom Message Format - Default Value `${message}`

reifl opened the pull request:
Added support to customize the Message Layout.
Default is `${message}` so the behaviour stays the same.

snakefoot commented on src/NLog.Layouts.GelfLayout/GelfConverter.cs:
```diff
@@ -24,10 +24,10 @@ public class GelfConverter
         private string _hostName;
 
-        public void ConvertToGelfMessage(JsonWriter jsonWriter, LogEventInfo logEventInfo, IGelfConverterOptions converterOptions)
+        public void ConvertToGelfMessage(JsonWriter jsonWriter, LogEventInfo logEventInfo, IGelfConverterOptions converterOptions, Layout messageLayout)
```
Use the `IGelfConverterOptions converterOptions` instead of adding a new parameter.

snakefoot commented:
Maybe add support for customizing both `ShortMessage` and `FullMessage` ?

https://archivedocs.graylog.org/en/latest/pages/gelf.html#gelf-payload-specification

reifl commented:
Changed to use the IGelfConverterOptions instead of the new parameter and removed the new parameter.
Added support to customize ShortMessage and FullMessage.

snakefoot commented on src/NLog.Layouts.GelfLayout/GelfConverter.cs:
```diff
@@ -27,14 +27,14 @@ public class GelfConverter
         {
             //Retrieve the formatted message from LogEventInfo
-            var logEventMessage = logEventInfo.FormattedMessage ?? string.Empty;
+            var logEventMessage = converterOptions.FullMessageLayout.Render(logEventInfo);
```
Could it become conditional so:
```
var fullMessage = converterOptions.FullMessageLayout?.Render(logEventInfo) ?? string.Empty;
```

snakefoot commented on src/NLog.Layouts.GelfLayout/GelfConverter.cs:
```diff
@@ -27,14 +27,14 @@ public class GelfConverter
 
             //Figure out the short message
-            var shortMessage = logEventMessage;
+            var shortMessage = converterOptions.ShortMessageLayout.Render(logEventInfo);
```
Could it become conditional so:
```
var shortMessage = converterOptions.ShortMessageLayout?.Render(logEventInfo) ?? string.Empty;
```

reifl commented:
fixed the possible NullReferenceException when the Render Function returns no string

snakefoot commented on src/NLog.Layouts.GelfLayout/GelfConverter.cs:
```diff
@@ -27,14 +27,14 @@ public class GelfConverter
         {
             //Retrieve the formatted message from LogEventInfo
-            var logEventMessage = logEventInfo.FormattedMessage ?? string.Empty;
+            var logEventMessage = converterOptions.FullMessageLayout.Render(logEventInfo) ?? string.Empty;
```
Can this become:
```
converterOptions.FullMessageLayout?.Render(logEventInfo) ?? string.Empty;
```

snakefoot commented on src/NLog.Layouts.GelfLayout/GelfConverter.cs:
```diff
@@ -27,14 +27,14 @@ public class GelfConverter
 
             //Figure out the short message
-            var shortMessage = logEventMessage;
+            var shortMessage = converterOptions.ShortMessageLayout.Render(logEventInfo) ?? string.Empty;
```
Can this become:
```
converterOptions.ShortMessageLayout?.Render(logEventInfo) ?? string.Empty;
```

snakefoot commented:
@tomikiss Looks good to me

tomikiss commented:
Thanks for the contribution @reifl and the review @snakefoot. I will try to release this in the coming days.

snakefoot commented:
@tomikiss Created a small adjustment #32

snakefoot commented:
@tomikiss Any progress on NLog.GelfLayout ver. 1.5 ?

tomikiss commented:
@snakefoot  Pushing this today evening.

snakefoot commented:
Thank you for the nuget-release - https://www.nuget.org/packages/NLog.GelfLayout/1.5.0

- Introduced `HostName` option to override GELF Message Host-field. Default = `${hostname}`
- Introduced `FullMessage` option to override GELF Full-Message-field (Max Length = 16383). Default = `${message}`
- Introduced `ShortMessage` option to override GELF Short-Message-field (Max Length = 250). Default = `${message}`
